express-notes-api/index.js

```diff
@@ -0,0 +1,130 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const jsonNotes = require('./data.json');
+
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  for (const eachNote in jsonNotes) {
+    const outer = jsonNotes[eachNote];
+    for (const key in outer) {
+      notesArray.push(outer[key]);
+    }
+  }
+  res.json(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const idNum = parseInt(req.params.id);
+  var flag;
+  if (idNum < 0) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  } else if (idNum > 0) {
+    for (const eachNote in jsonNotes) {
+      const notesNum = jsonNotes[eachNote];
+      for (const x in notesNum) {
+        var noteId = notesNum[x];
+        if (idNum === noteId.id) {
+          res.status(200).send(noteId);
+          flag = true;
+        }
+      }
+    }
```

Comment:
Instead of using a loop, you can check if `jsonNotes` at `idNum` is truthy or not. if not, then send the 404 and otherwise send the object.

```js
if (!jsonNotes[idNum]) {
```